### PR TITLE
Ports check TKE subroutine

### PR DESCRIPTION
**Pull request by singhbalwinder**
`check_tke` is ported in C++/Kokkos and a BFB test is added.

*Review this PR by hiding white space changes

### CODE REVIEW

File: components/scream/src/physics/shoc/shoc_check_tke.cpp [hunk 1]
```diff
@@ -0,0 +1,15 @@
+#include "shoc_check_tke_impl.hpp"
+#include "share/scream_types.hpp"
+
+namespace scream {
+  namespace shoc {
+
+    /*
+     * Explicit instantiation for doing vertflux on Reals using the
```
Comment (singhbalwinder) [inline components/scream/src/physics/shoc/shoc_check_tke.cpp:8]: I will fix this comment

**Review by ambrad — APPROVED** [review]

**Review by xyuan — APPROVED** [review]

**Review by jeff-cohere — APPROVED** [review]

File: components/cam/src/physics/cam/shoc.F90 [hunk 2]
```diff
@@ -3076,12 +3080,21 @@ subroutine check_tke(&
...
+   endif
+#endif
+
   do k=1,nlev
     do i=1,shcol
       tke(i,k)=max(mintke,tke(i,k))
     enddo
   enddo
 
+  return
```
Comment (jgfouca) [inline components/cam/src/physics/cam/shoc.F90:3097]: This return isn't needed.

File: components/scream/src/physics/shoc/shoc_check_tke.cpp [hunk 3]
```diff
@@ -0,0 +1,14 @@
+#include "shoc_check_tke_impl.hpp"
+#include "share/scream_types.hpp"
+
+namespace scream {
+  namespace shoc {
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/shoc_check_tke.cpp:5]: Namespaces should not increase the indent level.

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp [hunk 4]
```diff
@@ -818,6 +818,42 @@ void update_host_dse_f(Int shcol, Int nlev, Real* thlm, Real* shoc_ql, Real* exn
...
+
+  Kokkos::Array<view_2d, 1> temp_2d_d;
+
+  // Sync to device
+  ekat::pack::host_to_device({tke}, shcol, nlev, temp_2d_d, true);
+
+  view_2d
+    tke_d(temp_2d_d[0]);
+
+ const Int nk_pack = ekat::pack::npack<Spack>(nlev);
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/shoc_functions_f90.cpp:840]: You went from 2-space indent to 1.

**Review by jgfouca — CHANGES_REQUESTED** [review]
Some formatting cleans are needed, otherwise looks good.

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp (hunk 4 above)
Comment (singhbalwinder) [reply components/scream/src/physics/shoc/shoc_functions_f90.cpp:840]: Thanks @jgfouca for noticing these indenting issues.

File: components/scream/src/physics/shoc/shoc_check_tke.cpp [hunk 5]
```diff
@@ -0,0 +1,14 @@
+#include "shoc_check_tke_impl.hpp"
+#include "share/scream_types.hpp"
+
+namespace scream {
+namespace shoc {
+
+    /*
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/shoc_check_tke.cpp:7]: Still need to deindent these lines.

File: components/scream/src/physics/shoc/shoc_functions_f90.cpp [hunk 6]
```diff
@@ -884,5 +884,37 @@ void shoc_pblintd_init_pot_f(Int shcol, Int nlev, Real *thl, Real* ql, Real* q,
...
+
+  Kokkos::Array<view_2d, 1> temp_2d_d;
+
+  // Sync to device
+  ekat::host_to_device({tke}, shcol, nlev, temp_2d_d, true);
+
+  view_2d
+    tke_d(temp_2d_d[0]);
+
+ const Int nk_pack = ekat::npack<Spack>(nlev);
```
Comment (jgfouca) [inline components/scream/src/physics/shoc/shoc_functions_f90.cpp:904]: Still only see 1 space of indent here.

**Comment by singhbalwinder** [comment]
I have resolved all the issues but AT is failing tests which it should not. I have made it WIP until I figure that out.

**Comment by singhbalwinder** [comment]
All tests are passing now. @jgfouca : Would you please review again to see if everything looks good? Thanks!

**Review by jgfouca — APPROVED** [review]

**Pull request merged**